src/encoded/types/experiment.py

```diff
@@ -543,3 +597,45 @@ def __ac_local_roles__(self):
         root = find_root(self)
         experiment = root.get_by_uuid(properties['experiment'])
         return experiment.__ac_local_roles__()
+
+    @calculated_property(schema={
+        "title": "Libraries",
+        "description": "The nucleic acid libraries used in this replicate.",
+        "type": "array",
+        "uniqueItems": True,
+        "items": {
+            "title": "Library",
+            "description": "The nucleic acid library used in this replicate.",
+            "comment": "See library.json for available identifiers.",
+            "type": "string",
+            "linkTo": "Library"
+        }
+    })
+    def libraries(self, status, biological_replicate_number,
+                  technical_replicate_number):
+        if status == 'deleted':
+            return []
+        # Use root.get_by_uuid instead of embed to get reverse link
+        # specifically. This helps avoid infinite loop since calculated
+        # properties of experiment need to embed replicate.
+        properties = self.upgrade_properties()
+        root = find_root(self)
+        experiment = root.get_by_uuid(properties['experiment'])
+        libraries = set()
+        for rep_uuid in experiment.get_rev_links('replicates'):
+            rep_props = root.get_by_uuid(rep_uuid).upgrade_properties()
+            # Only care (check and add to the list) about non-deleted technical
+            # replicates of this replicate, meaning belonging to the same
+            # biological replicate.
+            if (rep_props['biological_replicate_number'] != biological_replicate_number
+                or rep_props['status'] == 'deleted'):
+                continue
+            if rep_props['technical_replicate_number'] < technical_replicate_number:
+                # Found smaller technical replicate, libraries will be
```

Comment:
Why?

Reply:
Overall, we use calculation to link all technical replicates of a biological replicate to only the first non deleted replicate. In the future, when we upgrade, all other technical replicates will be "deleted" and only this one will have the submittable property "libraries" filled.